### PR TITLE
DBAAS-1167: parse v1alpha1 provider inventory and connection spec and status

### PR DESCRIPTION
**Pull request by xieshenzh**
## Description

Parse v1alpha1 provider inventory and connection spec and status.
Jira: https://issues.redhat.com/browse/DBAAS-1167

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist (Definition of Done)

- [ ] All acceptance criteria specified in JIRA or in issue number have been completed
- [ ] Unit tests added that prove the fix is effective or the feature works 
- [ ] Documentation added for the feature
- [ ] all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer

### CODE REVIEW

File: controllers/dbaas_reconciler.go [hunk 1]
```diff
@@ -313,6 +315,14 @@ func (r *DBaaSReconciler) checkInventory(ctx context.Context, inventoryRef v1bet
 	return
 }
 
+func (r *DBaaSReconciler) getProviderSpecStatusVersion(provider *v1beta1.DBaaSProvider) *schema.GroupVersion {
+	if provider.GetDBaaSAPIGroupVersion().String() == v1alpha1.GroupVersion.String() &&
+		provider.Name != RDS_PROVIDER {
```
Comment (redhatHameed) [inline controllers/dbaas_reconciler.go:320]: just wondering do required to add !CRDB

File: controllers/dbaas_reconciler_test.go [hunk 2]
```diff
@@ -192,7 +192,7 @@ var _ = Describe("Watch DBaaS provider Object", func() {
 		owner := &v1beta1.DBaaSInventory{}
 		spyController := newSpyController(nil)
 
-		err := dRec.watchDBaaSProviderObject(spyController, owner, "test-kind")
+		err := dRec.watchDBaaSProviderObject(spyController, owner, "test-kind", &v1alpha1.GroupVersion)
```
Comment (jianrongzhang89) [inline controllers/dbaas_reconciler_test.go:195]: why do you watch v1alpha1?

File: controllers/dbaas_reconciler.go (hunk 1 above)
Comment (xieshenzh) [reply controllers/dbaas_reconciler.go:320]: No need.
CRDB is on v1beta1, it also has v1alpha1
RDS is on v1alpa1, but its spec and status on v1beta1

File: controllers/dbaas_reconciler_test.go (hunk 2 above)
Comment (xieshenzh) [reply controllers/dbaas_reconciler_test.go:195]: It is a test.

**Review by tchughesiv — APPROVED** [review]
lgtm

**Review by redhatHameed — APPROVED** [review]

**Pull request merged**